hub/core/chunk_engine.py

```diff
@@ -801,6 +803,7 @@ def extend(
                         link_callback(s, flat=True)
 
         else:
+            assert samples is not None
```

Comment:
will I get to this assert if I do ds.abc.extend(None)? If so then it should probably be an exception